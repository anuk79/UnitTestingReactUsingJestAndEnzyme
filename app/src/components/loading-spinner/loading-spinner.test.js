
// vendors
import React from "react";
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";

// components
import LoadingSpinner from './loading-spinner';

describe("LoadingSpinner component", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<LoadingSpinner customClassName="test-class"/>);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('div').length).toBe(1);
        expect(wrapper.find('div').text()).toBe('...loading');
        expect(wrapper.find('div').prop('className')).toBe('test-class');
    });
});