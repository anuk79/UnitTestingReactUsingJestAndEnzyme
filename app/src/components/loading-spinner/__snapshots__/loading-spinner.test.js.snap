// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component should render correctly 1`] = `
<div
  className="test-class"
>
  ...loading
</div>
`;
